arcgis-ios-sdk-samples/Augmented reality/View hidden infrastructure in AR/ViewHiddenInfrastructureARViewController.swift

```diff
@@ -0,0 +1,271 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ARKit
+import ArcGIS
+import ArcGISToolkit
+
+class ViewHiddenInfrastructureARViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The label to display AR status.
+    @IBOutlet var statusLabel: UILabel!
+    /// A segmented control to choose between roaming and local mode.
+    @IBOutlet var realScaleModePicker: UISegmentedControl!
+    /// The `ArcGISARView` managed by the view controller.
+    @IBOutlet var arView: ArcGISARView! {
+        didSet {
+            // Configure scene view.
+            let sceneView = arView.sceneView
+            sceneView.scene = makeScene()
+            // Configure overlays.
+            shadowsOverlay = makeShadowsOverlay(with: pipeGraphics)
+            leadersOverlay = makeLeadersOverlay(with: pipeGraphics)
+            sceneView.graphicsOverlays.addObjects(from: [makePipesOverlay(with: pipeGraphics), shadowsOverlay!, leadersOverlay!])
+            // Turn the space and atmosphere effects on for an immersive experience.
+            sceneView.spaceEffect = .transparent
+            sceneView.atmosphereEffect = .none
+            // Configure location data source.
+            arView.locationDataSource = AGSCLLocationDataSource()
+            arView.arSCNViewDelegate = self
+        }
+    }
+    
+    // MARK: Properties
+    
+    /// The graphics for pipe infrastructure passed from pipe placer view controller.
+    var pipeGraphics = [AGSGraphic]()
+    /// The elevation source with elevation service URL.
+    let elevationSource = AGSArcGISTiledElevationSource(url: URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!)
+    /// The elevation surface set to the base surface of the scene.
+    let elevationSurface = AGSSurface()
+    
+    /// A reference to the shadows overlay, to show ground level shadows of the underground pipes.
+    var shadowsOverlay: AGSGraphicsOverlay!
+    /// A reference to the leaders overlay, to show leader lines between ground and pipes.
+    var leadersOverlay: AGSGraphicsOverlay!
+    
+    // MARK: Methods
+    
+    /// Create a scene.
+    ///
+    /// - Returns: A new `AGSScene` object.
+    func makeScene() -> AGSScene {
+        let scene = AGSScene(basemapType: .imagery)
+        elevationSurface.navigationConstraint = .none
+        elevationSurface.opacity = 0
+        elevationSurface.elevationSources = [elevationSource]
+        scene.baseSurface = elevationSurface
+        return scene
+    }
+    
+    /// Create a graphic overlay for pipes and add graphics to it.
+    ///
+    /// - Parameter pipeGraphics: The graphics of the pipes.
+    /// - Returns: An `AGSGraphicsOverlay` object.
+    func makePipesOverlay(with pipeGraphics: [AGSGraphic]) -> AGSGraphicsOverlay {
```

Comment:
According to the API Design Guidelines, the label of the `pipeGraphics` parameter should just be `pipeGraphics`:

![image](https://user-images.githubusercontent.com/2257493/95477459-36ae9c00-093d-11eb-96fa-6022f9cbdbb8.png)

Reply:
Thanks. I'll need to refresh my memories on API design guidelines. 😄